lib/foreman_maintain/concerns/downstream.rb

```diff
@@ -46,8 +48,25 @@ def fm_pkg_and_cmd_name
         %w[satellite-maintain satellite-maintain]
       end
 
+      def satellite_maintain_target_version
+        satellite_maintain_config['current_sat_version']
+      end
+
+      def satellite_upgrade_allowed?
+        current_minor_version == satellite_maintain_config['previous_sat_version'] || \
+          ForemanMaintain.upgrade_in_progress == satellite_maintain_config['current_sat_version']
+      end
+
       private
 
+      def satellite_maintain_config
+        if File.exist?(SATELLITE_MAINTAIN_CONFIG)
+          YAML.load_file(SATELLITE_MAINTAIN_CONFIG)
+        else
+          {}
```

Comment:
This is a good sane default, but I wondering if we shouldn't throw a more in-yo-face error if the config file is missing as that indicates the system is likely in a bad state, something is missing and you can't proceed.